test/e2e/tests/eigen_test.go

```diff
@@ -0,0 +1,560 @@
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/test/e2e/types"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	omniDeployPk = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+	elDeployPk   = "0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6" // anvil account 9
+
+	operator1MetaDataURI = "https://www.operator1.com"
+	operator2MetaDataURI = "https://www.operator2.com"
+
+	MinStateForOperatorInEigenLayer = 10
+	MaxNumberOfOperators            = 200
+)
+
+var (
+	zeroAddr  = common.HexToAddress("0x0000000000000000000000000000000000000000")
+	omniDepPk = mustHexToKey(omniDeployPk)
+	elDepPk   = mustHexToKey(elDeployPk)
+)
+
+func TestEigenAndOmniAVS(t *testing.T) {
+	testAVS(t, func(t *testing.T, avs AVS, deployInfo map[types.ContractName]types.DeployInfo) {
+		t.Helper()
+		ctx := context.Background()
+
+		delMgrAddr := deployInfo[types.ContractELDelegationManager].Address
+		stratManAddr := deployInfo[types.ContractELStrategyManager].Address
+		wethStratAddr := deployInfo[types.ContractELWETHStrategy].Address
+		wethTokenAddr := deployInfo[types.ContractELWETH].Address
+		omniAVSAddr := deployInfo[types.ContractOmniAVS].Address
+
+		// create new operators, delegators and fund them with ETH
+		operator1Addr, opr1Addr, opr1Pk := createAccount(t, ctx, avs.Client)
+		operator2Addr, opr2Addr, opr2Pk := createAccount(t, ctx, avs.Client)
+		_, del1Addr, del1Pk := createAccount(t, ctx, avs.Client)
+		_, del2Addr, del2Pk := createAccount(t, ctx, avs.Client)
+
+		// check if contracts are deployed and configured properly
+		checkIfContractsAreDeployed(t, ctx, avs, deployInfo)
+		checkOmniAVSConfigurations(t, ctx, avs)
+
+		// register operators with EigenLayer
+		checkRegisteringOperatorToEL(t, ctx, avs, delMgrAddr, opr1Pk, operator1Addr, operator1MetaDataURI)
+		checkRegisteringOperatorToEL(t, ctx, avs, delMgrAddr, opr2Pk, operator2Addr, operator2MetaDataURI)
+
+		// add WETH funding to operators and delegators
+		checkFundingWETHToAccounts(t, ctx, avs, 1000, opr1Addr, opr2Addr, del1Addr, del2Addr)
+
+		// register operators to omni AVS with a stake more than minimum stake
+		whiteListStrategy(t, ctx, avs, wethStratAddr)
+		checkRegisteringOperatorsWithAVS(t, ctx, avs, stratManAddr, wethStratAddr, wethTokenAddr, omniAVSAddr,
+			opr1Addr, opr1Pk, 100)
+		checkRegisteringOperatorsWithAVS(t, ctx, avs, stratManAddr, wethStratAddr, wethTokenAddr, omniAVSAddr,
+			opr2Addr, opr2Pk, 100)
+
+		// delegate stake to operators and check their balance
+		delegateWETHToOperator(t, ctx, avs, del1Addr, del1Pk, opr1Addr)
+		delegateWETHToStrategy(t, ctx, avs, del1Pk, stratManAddr, wethStratAddr, wethTokenAddr, 50)
+		checkOetOperatorBalance(t, ctx, avs, opr1Addr, opr1Pk)
+
+		delegateWETHToOperator(t, ctx, avs, del2Addr, del2Pk, opr2Addr)
+		delegateWETHToStrategy(t, ctx, avs, del2Pk, stratManAddr, wethStratAddr, wethTokenAddr, 50)
+		checkOetOperatorBalance(t, ctx, avs, opr2Addr, opr2Pk)
+
```

Comment:
add TODOs for undelegation, deregistration of operators

See forge tests for example of all flows we need to test